sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetConverter.scala

```diff
@@ -426,6 +423,33 @@ private[parquet] class CatalystPrimitiveConverter(
     parent.updateLong(fieldIndex, value)
 }
 
+/**
+ * A `parquet.io.api.PrimitiveConverter` that converts Parquet Binary to Catalyst String.
+ * Supports dictionaries to reduce Binary to String conversion overhead.
+ *
+ * Follows pattern in Parquet of using dictionaries, where supported, for String conversion.
+ *
+ * @param parent The parent group converter.
+ * @param fieldIndex The index inside the record.
+ */
+private[parquet] class CatalystPrimitiveStringConverter(parent: CatalystConverter, fieldIndex: Int)
+  extends CatalystPrimitiveConverter(parent, fieldIndex) {
+
+  private var dict: Array[String] = null
```

Comment:
@squito JIT doesn't always take care of it. The main thing is because it is so easy to turn something into private[this], it is better to turn it into it most of the time.

And no, private is very useful too. You might want a private field to a class, but you'd want another instance of the same class to access it. For example, in equals method.